gpt/summary.py

```diff
@@ -0,0 +1,531 @@
+"""Script to summarise court transcripts using GPT-API."""
+from openai import OpenAI
+from dotenv import load_dotenv
+import json
+import time
+
+
+load_dotenv()
+openai = OpenAI()
+
+def get_extract_headings_prompt() -> str:
+    return """You are a UK legal data extraction assistant.
+    You will be given a list of headers from a court transcript.
+    You must return a list of headers that you believe will contain the necessary information in the transcript to deduce the following:
+    Summary: [a concise description of what the hearing was about, maximum 1000 characters]
+    Ruling: [which party the court ruled in favour of (one word answer e.g. Defendant)]
+    Anomalies: [whether anything irregular happened in the context of a normal court hearing]
+    """
```

Comment:
Might be worth specifying how the list should be structured, for example:


```suggestion
def get_extract_headings_prompt() -> str:
    return """You are a UK legal data extraction assistant.
    You will be given a list of headers from a court transcript.
    You must return a list of headers that you believe will contain the necessary information in the transcript to deduce the following:
    Summary: [a concise description of what the hearing was about, maximum 1000 characters]
    Ruling: [which party the court ruled in favour of (one word answer e.g. Defendant)]
    Anomalies: [whether anything irregular happened in the context of a normal court hearing]
    Give the list in the following format: "'heading1','heading2','heading3'"
    """
```

This way it'll be easier to break up the response into corresponding headings.

Reply:
Great suggestion! The prompts will need to be fine-tuned as we test the output and this is a good idea for how to improve it.